fjord/feedback/static/js/fxos_feedback.js

```diff
@@ -1,130 +1,173 @@
-        storageRemoveItem('description');
-    }
-
-    /*
-     * Infers the FxOS version from the Gecko version in the UA.
-     *
-     * If this can't find a gecko version in the ua, then it returns
-     * null. If this can find a gecko version in the ua, then it tries
-     * to map the gecko version to a FxOS version and return that. If it
-     * can't, then it returns the Gecko/version.
-     *
-     * Returns: null, FxOS version or Gecko versoin
-     */
-    function inferFxosVersion() {
-        var gecko_re = /Gecko\/([^\s]+)/,
-            possible_version = navigator.userAgent.match(gecko_re);
-
-        if (possible_version != null) {
-            possible_version = possible_version[1];
-            if (typeof(GECKO_TO_FXOS[possible_version]) !== 'undefined') {
-                possible_version = GECKO_TO_FXOS[possible_version];
-            } else {
-                possible_version = 'Gecko/' + possible_version;
+
+        function storageItem(key) {
+            if (window.localStorage) {
+                return localStorage[key];
             }
         }
 
-        return possible_version;
-    }
+        window.onpopstate = function(ev) {
+            // Switches the the appropriate card
+            var pageId = ev.state ? ev.state.page : 'intro';
+            cards.changeCard(pageId, false);
+        };
 
-    function init() {
-        $('#intro').on('click', 'button', function() {
-            var happy = $(this).hasClass('happy');
+        function formReset() {
+            $('#email-ok').prop('checked', false);
+            $('#description').val('');
+            fjord.countRemaining($('#description'));
 
-            $('#moreinfo')
-                .removeClass(happy ? 'sad' : 'happy')
-                .addClass(happy ? 'happy' : 'sad');
-            $('#happy').val(happy ? '1' : '0');
-            xdeck.nextCard();
-        });
+            storageRemoveItem('emailok');
+            storageRemoveItem('description');
+        }
 
-        $('button.back').on('click', function() {
-            xdeck.previousCard();
+        /**
+         * Selects happy/sad value and switches to the next card.
+         *
+         * @param {integer} val: 1 if happy, 0 if sad
+         */
+        function selectHappySad(val) {
+            if (val === 1) {
+                $('.happy').show();
+                $('.sad').hide();
+            } else {
+                $('.sad').show();
+                $('.happy').hide();
+            }
+            $('#happy').val(val);
+            // The happy/sad card is the intro card. This changes to
+            // the next card in the sequence.
+            cards.changeCard('forward');
+        }
+        $('#happy-button').click(function(ev) {
+            selectHappySad(1);
+            return false;
+        });
+        $('#happy-button').keypress(function(ev) {
+            if ((ev.keyCode || ev.which) === '13') {
+                $('#happy-button').click();
+            }
+        });
```

Comment:
nit: you could chain these two above. like:
```
    $('#happy-button')
        .click(function(ev) {
            selectHappySad(1);
            return false;
        })
        .keypress(function(ev) {
            if ((ev.keyCode || ev.which) === '13') {
                $('#happy-button').click();
            }
        });
```

It's mostly a styling preference but it saves a dom call to get the node.